favicon-from-sprite.js

```diff
@@ -0,0 +1,143 @@
+/*
+  ==============================
+  SPRITE FAVICON MOD (@mildmojo)
+  ==============================
+
+  Use one of your game sprites as the page favicon. It'll even animate if the
+  sprite has multiple frames!
+
+  HOW TO USE:
+    1. Copy-paste this script into a new script tag after the Bitsy source code.
+    2. Edit the top of the script to configure which sprite and colors it should
+       use for the favicon. By default, it will draw the player avatar sprite in
+       the first available palette's colors.
+
+  Version: 1.0
+  Bitsy Version: 4.5, 4.6
+  License: WTFPL (do WTF you want)
+*/
+
+// Give a hoot, don't pollute; encapsulate in an IIFE for isolation.
+(function(globals) {
+  'use strict';
+
+  // Configuration for favicon
+  // Sprite name as configured in editor. Defaults to player avatar.
+  var SPRITE_NAME = '';
+  // Palette name or number to draw colors from.
+  var PALETTE_ID = 0;
+  // Favicon background color in palette. 0 = BG, 1 = Tile, 2 = Sprite.
+  var BG_COLOR_NUM = 0;
+  // Favicon sprite color in palette. 0 = BG, 1 = Tile, 2 = Sprite.
+  var FG_COLOR_NUM = 2;
+  // Padding in pixels around sprite.
+  var PIXEL_PADDING = 2;
+  // Should the favicon have rounded corners? (Suggest margin 2px if rounding.)
+  var ROUNDED_CORNERS = true;
+  // Frame change interval if sprite is animated. Use `Infinity` to disable.
+  var FRAME_DELAY = 400;
+
+  var canvas = document.createElement('canvas');
+  canvas.width = 16+2*PIXEL_PADDING;
+  canvas.height = 16+2*PIXEL_PADDING;
+  var ctx = canvas.getContext('2d');
+  var faviconLoop;
+
+  var _load_game = load_game;
+  globals.load_game = function(game_data, startWithTitle) {
+    _load_game.apply(this, arguments);
+    startFaviconLoop();
+  };
+
+  var _stopGame = stopGame;
+  globals.stopGame = function() {
+    _stopGame.apply(this, arguments);
+    stopFaviconLoop();
+  }
+
+  function startFaviconLoop() {
+    var frameCount = getFrames(SPRITE_NAME).length;
+    if (frameCount === 1) {
+      return drawFavicon();
+    }
+
+    var i = 0;
+    faviconLoop = setInterval(function() {
```

Comment:
it would be better in the animated case to create the dataURLs for each frame ahead of time, and then only change the favicon link's href in the interval